### PR TITLE
configure.cmd user experience improvements by Trevor

### PR DESCRIPTION
**Pull request by ThFabba**
[CMAKE] Add some fool-proofing to configure.cmd related to MSVS:

-VSSolution switch now outputs to a dedicated folder if running from source directory.
-Give an error if the user is trying to mix ninja and VSSolution in the same output directory.
-In a MingW environment, give an error on VSSolution switch and give a warning on RTC switch.
-Fix -DNEW_STYLE_BUILD=0 option.
-Ensure the source path doesn't contain any spaces.
-Issue a warning if passed an invalid switch.
-Don't announce completion if there were errors.
-Make end message specific for VSSolution.
-Replace redundant exit commands with quit label, as suggested by HBelusca.

### CODE REVIEW

File: configure.cmd [hunk 1]
```diff
@@ -193,6 +229,21 @@ if "%REACTOS_SOURCE_DIR%" == "%CD%\" (
     cd %REACTOS_OUTPUT_PATH%
 )
 
+if "%VS_SOLUTION%" == "1" (
+
+    if exist build.ninja (
+        echo. && echo Error: This directory has already been configured for ninja.
+        echo An output folder configured for ninja can't be reconfigured for VSSolution.
+        echo Use an empty folder or delete the contents of this folder, then try again.
```
Comment (sanchaez) [inline configure.cmd]: We can force this.

File: configure.cmd [hunk 2]
```diff
@@ -250,11 +301,28 @@ if "%NEW_STYLE_BUILD%"=="0" (
...
+if %ERRORLEVEL% NEQ 0 (
+    goto quit
+)
+
+if "%CD_SAME_AS_SOURCE%" == "1" (
+    set ENDV= from %REACTOS_OUTPUT_PATH%
+)
+
+if "%VS_SOLUTION%" == "1" (
+    set ENDV= You can now open REACTOS.sln%ENDV%.
```
Comment (sanchaez) [inline configure.cmd]: ...or build it via `msbuild`.

File: configure.cmd (hunk 1 above)
Comment (learn-more) [reply configure.cmd]: No, people can have local modifications, we should not delete their shit.

Comment (sanchaez) [reply configure.cmd]: Sorry, what I really meant is a flag to do a force cleanup.

**Comment by gedmurphy** [comment]
This looks useful. Any reason it's not been merged yet?

**Comment by sanchaez** [comment]
@gedmurphy @ThFabba assigned himself on this task, you should probably ask him.

**Pull request merged**